WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/todaysstats/TodaysStatsCardBuilder.kt

```diff
@@ -1,21 +1,44 @@
 package org.wordpress.android.ui.mysite.cards.dashboard.todaysstats
 
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.dashboard.CardModel.TodaysStatsCardModel
+import org.wordpress.android.fluxc.store.dashboard.CardsStore.TodaysStatsCardError
+import org.wordpress.android.fluxc.store.dashboard.CardsStore.TodaysStatsCardErrorType
 import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.DashboardCards.DashboardCard.TodaysStatsCard
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.DashboardCards.DashboardCard.TodaysStatsCard.TodaysStatsCardWithData
 import org.wordpress.android.ui.mysite.MySiteCardAndItemBuilderParams.TodaysStatsCardBuilderParams
 import org.wordpress.android.ui.stats.refresh.utils.StatsUtils
+import org.wordpress.android.ui.utils.UiString.UiStringRes
 import org.wordpress.android.ui.utils.UiString.UiStringText
 import javax.inject.Inject
 
 class TodaysStatsCardBuilder @Inject constructor(
     private val statsUtils: StatsUtils
 ) {
     fun build(params: TodaysStatsCardBuilderParams) = params.todaysStatsCard?.let {
-        TodaysStatsCard(
-                views = statToUiString(it.views),
-                visitors = statToUiString(it.visitors),
-                likes = statToUiString(it.likes)
+        val error = it.error
+        if (error != null) {
+            createTodaysStatsCardWithError(error)
+        } else {
+            createTodaysStatsCardWithData(it)
+        }
+    }
+
+    private fun createTodaysStatsCardWithError(error: TodaysStatsCardError) = if (shouldShowError(error)) {
```

Comment:
Just a suggestion, can we separate out the logic of creating the `TodaysStatsCardError` and the logic of `shouldShowError(error)`. I believe the function `createTodaysStatsCardWithError` can have the responsibility of only creating `Today's stats` card. Something like this 

 ```
if (error != null) {
            handleError(error)
        } else {
            createTodaysStatsCardWithData(it)
        }
```

 ```
private fun handleError(error: TodaysStatsCardError) =
            if (shouldShowError(error)) createTodaysStatsCardWithError()
            else null

 private fun createTodaysStatsCardWithError() = TodaysStatsCard.Error(
            title = UiStringRes(R.string.my_site_todays_stat_card_title)
 )
```

Reply:
Suggestion accepted in 7c5f5f5. 🙂